### DISCUSSION TITLE
STM32H747XI Micropython Build Help

### DISCUSSION THREAD

**Discussion by AngeloFatica** [micropython/micropython#12920] [STM32 / Pyboard]
Hello everyone, 

I do not have much experience with building a custom micropython build for a specific MCU, Ive only used the F405 with the pyboard build. I was curious if I am able to use the micropython build for the Arduino Portenta H7 as its the same MCU as the one I will be using on a custom board (h747xi). If not, is there a place I can be pointed to with some help making a custom build as in what is needed to make a micropython build for a specific chip? Thanks.

**Comment by andrewleech** [comment]
Hi, yes in general when there's an existing board definition with the same MCU as one you're looking at using, that board is a great starting point. 
Copy that, then go through and remove anything that's referring to hardware you don't have, update any peripheral pins to match your usage and add in any new hardware bits you're using. You may need to double-check the crystal / clock setup you're using is the same as the example you've copied, but for many stm chips that's relatively standard.

There's also some examples here on how to set up a project with your own custom board profile _outside_ of the micropython tree; I use a project structure like this at work where my board profile and application python code is in my repo along with `micropython` as a submodule: 
https://github.com/micropython/micropython-example-boards

Note: in that repo there's a build Makefile in the board folder itself, so `cd` into the board folder you're interested in (eg. [CUSTOM_PYBV11](https://github.com/micropython/micropython-example-boards/tree/main/boards/CUSTOM_PYBV11)) and run `make`.
